src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java

```diff
@@ -125,4 +141,18 @@ private static String getTerminalFieldValue(FieldType field, Terminal terminal)
             default -> throw new PowsyblException(FIELD_AND_TYPE_NOT_IMPLEMENTED + " [" + field + ",terminal]");
         };
     }
+
+    private static String getSectionBasedFieldValue(FieldType field, ShuntCompensator shuntCompensator) {
+        double susceptancePerSection = shuntCompensator.getModel(ShuntCompensatorLinearModel.class).getBPerSection();
+        double voltageLevelPower = Math.pow(shuntCompensator.getTerminal().getVoltageLevel().getNominalV(), 2);
+
+        return switch (field) {
+            case SHUNT_COMPENSATOR_TYPE -> susceptancePerSection > 0 ? "CAPACITOR" : "REACTOR";
+            case MAX_Q_AT_NOMINAL_V -> String.valueOf(voltageLevelPower * susceptancePerSection * shuntCompensator.getMaximumSectionCount());
+            case SWITCHED_ON_Q_AT_NOMINAL_V -> String.valueOf(voltageLevelPower * susceptancePerSection * shuntCompensator.getSectionCount());
+            case MAX_SUSCEPTANCE -> String.valueOf(susceptancePerSection * shuntCompensator.getMaximumSectionCount());
+            case SWITCHED_ON_MAX_SUSCEPTANCE -> String.valueOf(susceptancePerSection * shuntCompensator.getSectionCount());
+            default -> throw new PowsyblException(FIELD_AND_TYPE_NOT_IMPLEMENTED + " [" + field + ",terminal]");
+        };
+    }
```

Comment:
Where this part come from please, to let me check ?

Reply:
It comes from network-modification-server in ShuntCompensatorModification.java (109 - 115)

Reply:
yes but you missed the Math.abs